core/server/src/log/logger.rs

```diff
@@ -417,6 +457,160 @@ impl Logging {
+        use std::fs;
+        use std::time::{SystemTime, UNIX_EPOCH};
+
+        tracing::debug!(
+            "Starting log cleanup for directory: {:?}, retention: {:?}, max_size: {} bytes",
+            logs_path,
+            retention,
+            max_size_bytes
+        );
+
+        let entries = match fs::read_dir(logs_path) {
+            Ok(entries) => entries,
+            Err(e) => {
+                tracing::warn!("Failed to read log directory {:?}: {}", logs_path, e);
+                return;
+            }
+        };
+
+        let mut file_entries = Vec::new();
+
+        for entry in entries.flatten() {
+            let metadata = match entry.metadata() {
+                Ok(metadata) => metadata,
+                Err(e) => {
+                    tracing::warn!("Failed to get metadata for {:?}: {}", entry.path(), e);
+                    continue;
+                }
+            };
+
+            if !metadata.is_file() {
+                continue;
+            }
+
+            let modified = match metadata.modified() {
+                Ok(modified) => modified,
+                Err(e) => {
+                    tracing::warn!(
+                        "Failed to get modification time for {:?}: {}",
+                        entry.path(),
+                        e
+                    );
+                    continue;
+                }
+            };
+
+            let elapsed = match modified.duration_since(UNIX_EPOCH) {
+                Ok(elapsed) => elapsed,
+                Err(e) => {
+                    tracing::warn!(
+                        "Failed to calculate elapsed time for {:?}: {}",
+                        entry.path(),
+                        e
+                    );
+                    continue;
+                }
+            };
+
+            let file_size = metadata.len();
+            file_entries.push((entry, modified, elapsed, file_size));
+        }
+
+        tracing::debug!(
+            "Processed {} log files from directory: {:?}",
+            file_entries.len(),
+            logs_path
+        );
+
+        let mut removed_files_count = 0;
+
+        if !retention.is_zero() {
+            let cutoff = match SystemTime::now().duration_since(UNIX_EPOCH) {
+                Ok(now) => now - retention,
+                Err(e) => {
+                    tracing::warn!("Failed to get current time: {}", e);
+                    return;
+                }
+            };
+
+            for (entry, _, elapsed, _) in &file_entries {
+                if *elapsed < cutoff {
+                    if let Err(e) = fs::remove_file(entry.path()) {
+                        tracing::warn!("Failed to remove old log file {:?}: {}", entry.path(), e);
+                    } else {
+                        tracing::debug!("Removed old log file: {:?}", entry.path());
+                        removed_files_count += 1;
+                    }
+                }
+            }
+        }
+
+        if max_size_bytes > 0 {
```

Comment:
To avoid the nested logic, this condition could be swapped with `if max_size_bytes == 0` + quick return from the function.